main.cc

```diff
@@ -62,9 +62,23 @@ int main(int argc, char *argv[]) {
     while (!input.eof()) {
         line_no++;
         getline(input, line);
+
+	// Strip any comments that start with ; to the end of the line
+	size_t comment_start = line.find(";");
+	if (comment_start != std::string::npos) {
+		line.replace(comment_start, line.length(), "");
+	}
+
+	// Now, strip away any trailing spaces
         while (!line.empty() && isspace(line[line.size()-1])) {
             line.resize(line.length()-1);
         }
+
+	// If the line is empty, then skip it
+	if (line.length() == 0) {
```

Comment:
If you formulate it as `if (line.empty()) { ... }` the intent might be even more clear.